File: backend/domain/entities/projeto.py
"""Entidade de domínio: Projeto."""

import uuid
from dataclasses import dataclass, field
from datetime import datetime


@dataclass
class Projeto:
    slug: str
    titulo: str
    descricao: str
    categoria: str
    imagem_capa: str
    ano: int
    id: uuid.UUID = field(default_factory=uuid.uuid4)
    imagens: list[str] = field(default_factory=list)
    ativo: bool = True
    criado_em: datetime = field(default_factory=datetime.utcnow)

File: frontend/core/domain/entities/Projeto.ts
export interface PlainProjeto {
  id: string;
  slug: string;
  titulo: string;
  descricao: string;
  categoria: string;
  imagemCapa: string;
  imagens: string[];
  ano: number;
  ativo: boolean;
  criadoEm: string;
}

export class Projeto {
  constructor(
    public readonly id: string,
    public readonly slug: string,
    public readonly titulo: string,
    public readonly descricao: string,
    public readonly categoria: string,
    public readonly imagemCapa: string,
    public readonly imagens: string[],
    public readonly ano: number,
    public readonly ativo: boolean,
    public readonly criadoEm: string
  ) {}
}
